src/Symfony/Component/Yaml/Parser.php

```diff
@@ -429,7 +429,24 @@ private function doParse($value, $flags)
 
                     foreach ($this->lines as $line) {
                         try {
-                            $parsedLine = $line;
+                            if (isset($line[0]) && ('"' === $line[0] || "'" === $line[0])) {
+                                $parsedLine = $line;
+                            } else {
+                                $parsedLine = Inline::parse($line, $flags, $this->refs);
+                            }
+
+                            // There is a special case where an empty array
+                            // followed by a comment causes us to try to parse
+                            // the value as a multi-line string. In this
+                            // instance, return the empty array
+                            if ([] === $parsedLine) {
```

Comment:
This does not work. If I change the {} to 1.1 ie. float then we get the parse exception.